parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -324,6 +346,92 @@ public static Statistics toParquetStatistics(
     return stats;
   }
 
+  public org.apache.parquet.column.statistics.Statistics fromParquetStatistics(
+      String createdBy, Statistics statistics, PrimitiveType type) {
+    SortOrder expectedOrder = isSignedOrderOkay(type) ? SortOrder.SIGNED : sortOrder(type);
+    return fromParquetStatisticsInternal(
+        createdBy, statistics, type.getPrimitiveTypeName(), expectedOrder);
+  }
+
+  enum SortOrder {
+    SIGNED,
+    UNSIGNED,
+    UNKNOWN
+  }
+
+  private boolean isSignedOrderOkay(PrimitiveType type) {
+    if (!useSignedStringMinMax) {
+      return false;
+    }
+
+    // even if the override is set, only return stats for string-ish types
+    if (type.getPrimitiveTypeName() != PrimitiveTypeName.BINARY) {
+      return false;
+    }
+    if (type.getOriginalType() == null) {
+      return true; // plain binary is okay
+    }
+    switch (type.getOriginalType()) {
+      case UTF8:
+      case ENUM:
+      case BSON:
+      case JSON:
+        return true;
+      default: // includes decimal
+        return false;
+    }
+  }
```

Comment:
Why not simply:
```
private boolean isMinMaxUsableFor(PrimitiveType type) {
 SortOrder expected = sortOrder(type);
 return (expected == SIGNED) || (expected == UNSIGNED && useSignedMinMaxForUnsigned);
}
```

the case for getOriginalType is already treated in sortOrder(). Why adding another layer here?